### DISCUSSION TITLE
Add "defer" attribute to injected scripts

### DISCUSSION THREAD

**Discussion by RoKrypto** [wintercms/winter#614] [Q&A] [answered]
Greetings. I have gone through the documentation to check if there's a function that allows me to set the "defer" attribute to all script tags that are injected through `{% scripts %}`. The only similar function I was able to find is `$this->addJs()`, but I'm required to add a publicly accessible asset to the injected page assets. Is there a way I can aim to the injected scripts only?

**Comment by LukeTowers** [comment, marked as answer]
You should be able to use the `system.assets.beforeAddAsset` event to do that: https://wintercms.com/docs/events/event/system.assets.beforeAddAsset

**Reply by RoKrypto** [reply]
Hi @LukeTowers . Thank you for replying. I'm not sure about doing things right. Could you check if the following code is the right way to go?

```php
<?php namespace DesignneerStudio\ScriptsDeferrer;

class Plugin extends \System\Classes\PluginBase
{
    public function pluginDetails()
    {
        return [
            'name' => 'Scripts Deferrer Plugin',
            'description' => 'Add the "defer" attribute to scripts injected by Twig tag {& scripts %}.',
            'author' => 'Designneer Studio',
            'icon' => 'icon-code'
        ];
    }

    public function boot()
    {
        Event::listen('system.assets.beforeAddAsset', function (string &$type, string &$path, array &$attributes) {
            if($type === 'js' || strpos($path, '.js')) {
                $attributes['defer'] = true;
            }
            return false;
        });
    }
}
```

**Reply by LukeTowers** [reply]
Looks about right, but don't return anything

**Reply by RoKrypto** [reply]
Hi @LukeTowers . Do you know the approach I should follow for the scripts injected by **{% framework %}**. What I just showed you doesn't seem to work. Any piece of advice?

**Reply by LukeTowers** [reply]
Your best bet would be to replace the use of `{% framework %}` with the following:

```twig
<script src="{{ ['@framework'] | theme }}" defer="true"></script>
```

**Reply by RoKrypto** [reply]
Cool! Thank you once again, @LukeTowers
